docs/data-structures/Singly-Linked-list.mdx

```diff
@@ -0,0 +1,589 @@
+    print("2. Insert Node at End")
+    print("3. Insert Node at Position")
+    print("4. Delete Node at Front")
+    print("5. Delete Node at End")
+    print("6. Delete Node at Position")
+    print("7. Print the List")
+    print("8. Count nodes in the List")
+    print("9. Exit")
+    print("--Use 1-based indexing--")
+    while(True):
+        ch = int(input("Enter your choice: "))
+        if ch == 1:
+            ndata = int(input("Enter the element to insert: "))
+            newLinkedList.insertNodeAtFront(ndata)
+        elif ch == 2:
+            ndata = int(input("Enter the element to insert: "))
+            newLinkedList.insertNodeAtEnd(ndata)
+        elif ch == 3:
+            ndata = int(input("Enter the element to insert: "))
+            ind = int(input("Enter index to insert this element: "))
+            newLinkedList.insertNodeAtPosition(ind,ndata)
+        elif ch == 4:
+            newLinkedList.deleteNodeAtFront()
+        elif ch == 5:
+            newLinkedList.deleteNodeAtEnd()
+        elif ch == 6:
+            ind = int(input("Enter index to delete element: "))
+            newLinkedList.deleteNodeAtPosition(ind)
+        elif ch == 7:
+            newLinkedList.traverseList()
+        elif ch == 8:
+            newLinkedList.countNodes()
+        elif ch == 9:
+            break
+        else:
+            print("Enter a valid choice (1-9)")
+
+```
+</TabItem>
+</Tabs>
+
+## Sample I/O
+
+### Sample Input
+```
+SINGLY LINKED LIST
+1. Insert Node at Front
+2. Insert Node at End
+3. Insert Node at Position
+4. Delete Node at Front
+5. Delete Node at End
+6. Delete Node at Position
+7. Print the List
+8. Count nodes in the List
+9. Exit
+Enter your choice: 1
+Enter the element to insert: 20
+Enter your choice: 2
+Enter the element to insert: 67
+Enter your choice: 1
+Enter the element to insert: 66
+Enter your choice: 2
+Enter the element to insert: 89
+Enter your choice: 1
+Enter the element to insert: 56
+Enter your choice: 5
+Enter your choice: 6
+Enter index to delete element: 2
+Enter your choice: 1
+Enter the element to insert: 45
+Enter your choice: 8
+4
+Enter your choice: 1
+Enter the element to insert: 43
+Enter your choice: 7
+```
+
+### Sample Output
+```
+43
+45
+56
+20
+67
+```
+## Complexity Analysis
+- Access
+    - **Average Case**: O (n)
+    - **Worst Case**: O (n)
+- Search 
+    - **Average Case**: O (n)
+    - **Worst Case**: O (n)
+- Insertion
+    - **Average Case**: O (1)
+    - **Worst Case**: O (1)
+- Deletion
+    - **Average Case**: O (1)
+    - **Worst Case**: O (1)
+
+- Space Complexity: O (n)
+
+## Credits
+
+- [Sangeeta Mishra](https://github.com/SangeetaMishra) for the [Python Implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/ds/SinglyLinkedList.py)
+- [Vatsal kesarwani](https://github.com/plazzy99) for the [C++ Implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Interpolation_Search.cpp)
```

Comment:
This implementation is by @sharur7 [here](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Interpolation_Search.cpp)